docs/source/layout.rst

```diff
@@ -0,0 +1,600 @@
+subsections of an :code:`ON` block with each other. If set to :code:`False`
+the original SQL would become:
+
+.. code-block:: sql
+
+   SELECT
+      a,
+      b
+   FROM my_table
+   JOIN another_table
+      ON condition1
+      AND condition2
+
+These can also be combined, so if :code:`indented_using_on` config is set to
+:code:`False`, and :code:`indented_on_contents` is also set to :code:`False`
+then the SQL would become:
+
+.. code-block:: sql
+
+   SELECT
+      a,
+      b
+   FROM my_table
+   JOIN another_table
+   ON condition1
+   AND condition2
+
+There is also a similar :code:`indented_ctes` config (defaulted to
+:code:`False`) which can be set to :code:`True` to enforce CTEs to be
+indented within the :code:`WITH` clause:
+
+.. code-block:: sql
+
+   WITH
+      some_cte AS (
+         SELECT 1 FROM table1
+      ),
+
+      some_other_cte AS (
+         SELECT 1 FROM table1
+      )
+
+   SELECT 1 FROM some_cte
+
+By default, *SQLFluff* aims to follow the indentation most common approach
+to indentation. However, if you have other versions of indentation which are
+supported by published style guides, then please submit an issue on GitHub
+to have that variation supported by *SQLFluff*.
+
+.. _layoutspacingconfig:
+
+Configuring layout and spacing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :code:`[sqlfluff:layout]` section of the config controls the treatment of
+spacing and line breaks across all rules. The syntax of this section is very
+expressive, however in normal use, only very small alterations should be
+necessary from the :ref:`defaultconfig`.
+
+The syntax of the section headings here select by *type*, which corresponds
+to the :code:`type` defined in the dialect. For example the following section
+applies to elements of the *type* :code:`comma`, i.e. :code:`,`.
+
+.. code-block:: cfg
+
+   [sqlfluff:layout:type:comma]
+   spacing_before = touch
+   line_position = trailing
+
+Within these configurable sections there are a few key elements which are
+available:
+
+*  **Spacing Elements**: :code:`spacing_before`, :code:`spacing_after` and
+   :code:`spacing_within`. For each of these options there are a few possible
+   settings:
+
+   *  The default spacing for all elements is :code:`single` unless otherwise
+      specified. In this state, elements will be spaced with a single space
+      character unless there is a line break between them.
+
+   *  The value of :code:`touch` allows line breaks, but if no line break is
+      present, then no space should be present. A great example of this is
+      the spacing before commas (as shown in the config above), where line
+      breaks may be allowed, but if not they should *touch* the element before.
+
+   *  The value of :code:`inline` is effectively the same as :code:`touch`
+      but in addition, no line breaks are allowed. This best illustrated
+      by the :code:`spacing_within` a qualified identifier like
```

Comment:
The wording here seems strange -- can you review? "by the XXX a qualified identifier"